### PR TITLE
[SD-1392] - Add Swagger API Docs For Menu and Pages CMS

### PR DESCRIPTION
**Pull request by MatthewKennedy**
THIS PR IS GOING TO BE SUPERSEDED.

- [ ] Add Swagger Docs for Menus
- [ ] Setup menu_item.icon access through API.
- [ ] Add swagger API Docs for CMS Pages

As it stands icon is not resized like image, this is intentional so that it can be used for svg's

### CODE REVIEW

File: api/app/controllers/spree/api/v2/platform/menus_controller.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@ module Platform
         class MenusController < ResourceController
           private
 
+          def resource
```
Comment (damianlegawiec) [inline api/app/controllers/spree/api/v2/platform/menus_controller.rb:8]: @MatthewKennedy let's add some request specs for this please

Comment (MatthewKennedy) [reply api/app/controllers/spree/api/v2/platform/menus_controller.rb:8]: No problem, just doing the swagger docs so will get the test written shortly.

Comment (MatthewKennedy) [reply api/app/controllers/spree/api/v2/platform/menus_controller.rb:8]: looking through the specs you can filter by location so this is not needed.

File: api/docs/v2/storefront/index.yaml [hunk 2]
```diff
@@ -94,18 +94,6 @@ paths:
                           state_code: NY
         '403':
           $ref: '#/components/responses/403Forbidden'
-          description: ''
```
Comment (damianlegawiec) [inline api/docs/v2/storefront/index.yaml:97]: @MatthewKennedy why are we removing this one?

File: api/docs/v2/storefront/index.yaml [hunk 3]
```diff
@@ -4486,6 +4577,153 @@ components:
...
+          $ref: '#/components/schemas/IconAttributes'
+    IconAttributes:
+      type: object
+      properties:
+        viewable_type:
+          type: string
+          example: Spree::MenuItem
+        viewable_id:
+          type: string
+          example: 22
```
Comment (damianlegawiec) [inline api/docs/v2/storefront/index.yaml:4723]: @MatthewKennedy I think we don't need these as attribute, relationship handles this well, no need for duplication

File: api/docs/v2/storefront/index.yaml (hunk 2 above)
Comment (MatthewKennedy) [reply api/docs/v2/storefront/index.yaml:97]: According to Swagger:

`Sibling values alongside $refs are ignored. To add properties to a $ref, wrap the $ref into allOf, or move the extra properties into the referenced definition (if applicable).`

Comment (MatthewKennedy) [reply api/docs/v2/storefront/index.yaml:97]: As far as I can tell from the output `$ref: '#/components/responses/403Forbidden'` is taking care of the description.

File: api/docs/v2/storefront/index.yaml (hunk 3 above)
Comment (MatthewKennedy) [reply api/docs/v2/storefront/index.yaml:4723]: I'll give the Menus one a good going over, As I was doing the pages I realised I need to go back over the menus again.

**Pull request closed without merging**